express-notes-api/index.js

```diff
@@ -0,0 +1,134 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+let noteObj = {};
+const fs = require('fs')
+
+app.listen(3000, () => {
+  console.log('Listening on port 3000!');
+})
+
+// 1) Clients can GET a list of notes.
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (prop in data.notes) {
+    notesArray.push(data.notes[prop]);
+  }
+  res.status(200).json(notesArray);
+})
+
+// 2) Clients can GET a single note by id.
+app.get('/api/notes/:id', (req, res) => {
+  const requestID = req.params.id;
+
+  // if requested id not a positive integer
+  if (isNaN(requestID) || Number(requestID) <= 0) {
+    noteObj = { error: 'id must be a positive integer' };
+    res.status(400).json(noteObj);
+    return;
+  };
```

Comment:
What about if `requestID` is a positive non-integer number (ex: `4.23`)?

Reply:
@TimelikeClosure
Update line26 with additional condition to check if it's an integer number.
![image](https://user-images.githubusercontent.com/76827487/131240964-0a144df9-335d-4f1c-80a8-7ae71d226393.png)